### PR TITLE
[Benchmark] Add INT8 PTQ pipeline and iGPU benchmark for ViT backbone

### LINKED ISSUE #35023: [Performance]: Missing reproducible INT8 quantization benchmark for vision-transformer backbones on Intel iGPU via NNCF
### OpenVINO Version

Master Branch

### Operating System

Windows System

### Device used for inference

iGPU

### OpenVINO installation

PyPi

### Programming Language

Python

### Hardware Architecture

x86 (64 bits)

### Model used

ViT-B/16

### Model quantization

No

### Target Platform

_No response_

### Performance issue description

There is no reproducible benchmark or documented pipeline for measuring the accuracy/latency tradeoff of INT8 post-training quantization (PTQ) applied to vision-transformer backbones on Intel iGPU using NNCF. Existing benchmarks and documentation focus primarily on CNN architectures (ResNet, MobileNet, YOLO) and CPU targets.

As VLA models for embodied intelligence (e.g. GR00T, Pi0) rely heavily on vision-transformer encoders, users deploying these systems on Intel Architecture lack a validated reference for applying NNCF INT8 quantization and measuring the resulting latency/accuracy impact on iGPU specifically. 

Expected outcome: a reproducible pipeline and benchmark showing FP32 vs INT8 latency and top-1 accuracy tradeoff for a ViT backbone compiled for Intel iGPU.

### Step-by-step reproduction

1. Install OpenVINO 2026.0 and NNCF
2. Load ViT-B/16 from Hugging Face and convert to OpenVINO IR via ov.convert_model()
3. Apply NNCF INT8 PTQ using nncf.quantize() with a calibration dataloader
4. Compile both FP32 and INT8 models for GPU device via core.compile_model(model, "GPU")
5. Benchmark inference latency on Intel iGPU using benchmark_app or a custom timing loop
6. Measure top-1 accuracy on ImageNet validation subset for both precisions
7. Observe no existing documented reference exists to validate these results against or follow as a standard pipeline

### Issue submission checklist

- [x] I'm reporting a performance issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

### PR DESCRIPTION
**Pull request by jayden1711**
### Details:

- Adds a new Python sample vit_int8_igpu_benchmark under samples/python/
- Demonstrates INT8 post-training quantization (PTQ) of a ViT-B/16 vision-transformer backbone using NNCF
- Benchmarks FP32 vs INT8 latency and top-1 accuracy on Intel iGPU via OpenVINO GPU plugin
- Addresses a gap in existing benchmarks which focus only on CNN architectures (ResNet, MobileNet, YOLO) and CPU targets
- As VLA models for embodied intelligence (e.g. GR00T, Pi0) rely on vision-transformer encoders, this sample provides a validated reference pipeline for practitioners deploying these systems on Intel Architecture
- Includes README.md with usage instructions, expected output, and results table

### Tickets:
 - Resolves #35023

### AI Assistance:
 - AI assistance used: no

### CODE REVIEW

File: samples/python/vit_int8_igpu_benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+[ INFO ]   FP32          18.43            0.8110
+[ INFO ]   INT8          9.17             0.8043
+[ INFO ] ==================================================
+```
+
+## References
+
+- [OpenVINO NNCF](https://github.com/openvinotoolkit/nncf)
+- [ViT-B/16 on Hugging Face](https://huggingface.co/google/vit-base-patch16-224)
+- Resolves [#YOUR_ISSUE_NUMBER](https://github.com/openvinotoolkit/openvino/issues/YOUR_ISSUE_NUMBER)
```
Comment (almilosz) [inline samples/python/vit_int8_igpu_benchmark/README.md:46]:
```suggestion
- Resolves [???](https://github.com/openvinotoolkit/openvino/issues/YOUR_ISSUE_NUMBER)
```

File: samples/python/vit_int8_igpu_benchmark/vit_int8_igpu_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This sample demonstrates INT8 post-training quantization (PTQ) of a
+vision-transformer (ViT) backbone using NNCF, and benchmarks FP32 vs INT8
+latency and top-1 accuracy on Intel iGPU.
```
Comment (almilosz) [inline samples/python/vit_int8_igpu_benchmark/vit_int8_igpu_benchmark.py:9]: Other benchmarks are in _samples/python/benchmark_ directory

File: samples/python/vit_int8_igpu_benchmark/vit_int8_igpu_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+Resolves: https://github.com/openvinotoolkit/openvino/issues/35023
+"""
+
+import logging as log
+import sys
+import time
+import numpy as np
+import openvino as ov
+import nncf
+import torch
```
Comment (almilosz) [inline samples/python/vit_int8_igpu_benchmark/vit_int8_igpu_benchmark.py:20]: _requirements.txt_ file is missing for this benchmark

**Review by almilosz — CHANGES_REQUESTED** [review]
I think this would fit better as an NNCF example or OpenVINO notebook, similar to the existing [MobileNet V2 PTQ example](https://github.com/openvinotoolkit/nncf/tree/develop/examples/post_training_quantization/openvino/mobilenet_v2 ) . In  [this](https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/pytorch-post-training-quantization-nncf/pytorch-post-training-quantization-nncf.ipynb) notebook there is also a part with INT8 model and FP32 model comparison

The proposed benchmark is quite model- and workflow-specific, so imo it's too specialized for the general OpenVINO benchmark samples.
